subsys/bluetooth/host/hci_core.c

```diff
@@ -1740,7 +1715,27 @@ int bt_unpair(u8_t id, const bt_addr_le_t *addr)
 	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
 		bt_gatt_clear(id, addr);
 	}
+}
+
+static void bt_unpair_with_key(const struct bt_bond_info *info, void *data)
```

Comment:
_with_key could be a bit mislead, especially with all the other uses of key in this context. Maybe `unpair_addr` or `unpair_remote`?